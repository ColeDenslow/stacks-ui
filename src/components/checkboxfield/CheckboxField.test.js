/* global describe, test, expect */
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import React from 'react';
import CheckboxField from './CheckboxField';

describe('CheckboxField', () => {
  test('CheckboxField renders with basic props', () => {
    const onChange = () => 'onChange';
    const onClick = () => 'onClick';

    const output = shallow(<CheckboxField
      id='test-id'
      className='html-class'
      name='name'
      value='test-string'
      label='checkboxfield-label'
      onChange={onChange}
      onClick={onClick}
    />);

    expect(toJson(output)).toMatchSnapshot();
  });

  test('CheckboxField renders without any props', () => {
    const output = shallow(<CheckboxField />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('CheckboxField renders checked', () => {
    const output = shallow(<CheckboxField checked />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('CheckboxField renders with helpText', () => {
    const output = shallow(<CheckboxField helpText='help-text' />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('CheckboxField renders with a success messageText', () => {
    const output = shallow(<CheckboxField
      messageText='message-text'
      messageType='success'
    />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('CheckboxField renders with a warning messageText', () => {
    const output = shallow(<CheckboxField
      messageText='message-text'
      messageType='warning'
    />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('CheckboxField renders with an alert messageText', () => {
    const output = shallow(<CheckboxField
      messageText='message-text'
      messageType='alert'
    />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('CheckboxField render disabled when disabled is true', () => {
    const output = shallow(<CheckboxField disabled />);
    expect(toJson(output)).toMatchSnapshot();
  });

  test('CheckboxField render additional props', () => {
    const output = shallow(<CheckboxField data-additional-prop='additional-prop' />);
    expect(toJson(output)).toMatchSnapshot();
  });
});
