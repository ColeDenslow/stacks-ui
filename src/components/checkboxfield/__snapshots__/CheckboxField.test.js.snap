// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField CheckboxField render additional props 1`] = `
<Field
  className="stx-checkbox-field__field"
>
  <div
    className="stx-checkbox-field__div"
  >
    <Checkbox
      className="stx-checkbox-field__checkbox"
      data-additional-prop="additional-prop"
      id="stxField9"
    />
    <label
      className="stx-checkbox-field__label"
      htmlFor="stxField9"
    />
  </div>
</Field>
`;

exports[`CheckboxField CheckboxField render disabled when disabled is true 1`] = `
<Field
  className="stx-checkbox-field__field"
>
  <div
    className="stx-checkbox-field__div"
  >
    <Checkbox
      className="stx-checkbox-field__checkbox"
      disabled={true}
      id="stxField8"
    />
    <label
      className="stx-checkbox-field__label"
      htmlFor="stxField8"
    />
  </div>
</Field>
`;

exports[`CheckboxField CheckboxField renders checked 1`] = `
<Field
  className="stx-checkbox-field__field"
>
  <div
    className="stx-checkbox-field__div"
  >
    <Checkbox
      checked={true}
      className="stx-checkbox-field__checkbox"
      id="stxField3"
    />
    <label
      className="stx-checkbox-field__label"
      htmlFor="stxField3"
    />
  </div>
</Field>
`;

exports[`CheckboxField CheckboxField renders with a success messageText 1`] = `
<Field
  className="stx-checkbox-field__field"
  messageText="message-text"
  messageType="success"
>
  <div
    className="stx-checkbox-field__div"
  >
    <Checkbox
      className="stx-checkbox-field__checkbox"
      id="stxField5"
    />
    <label
      className="stx-checkbox-field__label"
      htmlFor="stxField5"
    />
  </div>
</Field>
`;

exports[`CheckboxField CheckboxField renders with a warning messageText 1`] = `
<Field
  className="stx-checkbox-field__field"
  messageText="message-text"
  messageType="warning"
>
  <div
    className="stx-checkbox-field__div"
  >
    <Checkbox
      className="stx-checkbox-field__checkbox"
      id="stxField6"
    />
    <label
      className="stx-checkbox-field__label"
      htmlFor="stxField6"
    />
  </div>
</Field>
`;

exports[`CheckboxField CheckboxField renders with an alert messageText 1`] = `
<Field
  className="stx-checkbox-field__field"
  messageText="message-text"
  messageType="alert"
>
  <div
    className="stx-checkbox-field__div"
  >
    <Checkbox
      className="stx-checkbox-field__checkbox"
      id="stxField7"
    />
    <label
      className="stx-checkbox-field__label"
      htmlFor="stxField7"
    />
  </div>
</Field>
`;

exports[`CheckboxField CheckboxField renders with basic props 1`] = `
<Field
  className="stx-checkbox-field__field html-class"
>
  <div
    className="stx-checkbox-field__div"
  >
    <Checkbox
      className="stx-checkbox-field__checkbox"
      id="test-id"
      name="name"
      onChange={[Function]}
      onClick={[Function]}
      value="test-string"
    />
    <label
      className="stx-checkbox-field__label"
      htmlFor="test-id"
    >
      checkboxfield-label
    </label>
  </div>
</Field>
`;

exports[`CheckboxField CheckboxField renders with helpText 1`] = `
<Field
  className="stx-checkbox-field__field"
  helpText="help-text"
>
  <div
    className="stx-checkbox-field__div"
  >
    <Checkbox
      className="stx-checkbox-field__checkbox"
      id="stxField4"
    />
    <label
      className="stx-checkbox-field__label"
      htmlFor="stxField4"
    />
  </div>
</Field>
`;

exports[`CheckboxField CheckboxField renders without any props 1`] = `
<Field
  className="stx-checkbox-field__field"
>
  <div
    className="stx-checkbox-field__div"
  >
    <Checkbox
      className="stx-checkbox-field__checkbox"
      id="stxField2"
    />
    <label
      className="stx-checkbox-field__label"
      htmlFor="stxField2"
    />
  </div>
</Field>
`;
